crates/derivation/src/channel_bank.rs

```diff
@@ -0,0 +1,95 @@
+use super::channel::Channel;
+use super::frame::Frame;
+use core::prelude::*;
+
+use core::types::H128;
+use std::collections::{HashMap, VecDeque};
+
+const MAX_CHANNEL_BANK_SIZE: u64 = 100_000_000;
+
+#[derive(Default, Debug)]
+pub struct ChannelBank {
+	channels_map: HashMap<H128, Channel>,
+	channels_by_creation: VecDeque<H128>,
+}
+
+impl ChannelBank {
+	pub fn load_frame(&mut self, frame: Frame, l1_block: BlockID) {
+		assert!(
+			!self.peek().is_some_and(|c| c.is_ready()),
+			"Specs Violation: must pull data before loading more in the channel bank"
+		);
```

Comment:
The `assert` macro invokes `panic` at runtime, worth removing in the future with bubbled errors

Reply:
Intentional. There's not an easy way to otherwise enforce this rule